Scripts/Mac/Common-Functions.ps1

```diff
@@ -0,0 +1,153 @@
+<#
+.SYNOPSIS
+Generates the text for GVFS.Installer.Mac nuspec file from the provided inputs
+#>
+function Write-Nuspec
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [string]
+        $PackageVersion,
+
+        [Parameter(Mandatory)]
+        [string]
+        $GvfsInstallerPkg,
+
+        [Parameter(Mandatory)]
+        [string]
+        $GitInstallerPkg,
+
+        [Parameter(Mandatory)]
+        [string]
+        $GcmInstallerPkg
+    )
+
+     $template =
+     "<?xml version=""1.0""?>
+     <package xmlns=""http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd"">
+       <metadata>
+         <id>GVFS.Installers.Mac</id>
+         <version>$PackageVersion</version>
+         <authors>Microsoft</authors>
+         <requireLicenseAcceptance>false</requireLicenseAcceptance>
+         <description>GVFS and G4M Mac installers</description>
+       </metadata>
+       <files>
+         <file src=""$GvfsInstallerPkg"" target=""GVFS"" />
+         <file src=""$GitInstallerPkg"" target=""G4M"" />
+         <file src=""$GcmInstallerPkg"" target=""GCM"" />
+       </files>
+     </package>"
+
+     return $template
+}
+
+<#
+.DESCRIPTION
+Downloads the specified version of GCM Core to the specified DownloadLocation
+#>
+function Download-GCM
+{
+    param
+    (
+        [Parameter(Mandatory)]
+        [string]
+        $GcmVersion,
+
+        [Parameter(Mandatory)]
+        [string]
+        $DownloadLocation
+    )
+
+    $packageName = "gcmcore-osx-$GcmVersion.pkg"
+    $url = "https://github.com/microsoft/Git-Credential-Manager-Core/releases/download/v2.0.33-beta/$packageName"
```

Comment:
REVIEW (for @jamill): This looks odd - why is v2.0.33-beta in the hard coded URL. One the one hand - this is temporary code for how we consume GCM now, and will be removed as we consume GCM from an internal NuGet feed.